pippo-core/src/main/java/ro/pippo/core/route/ClasspathResourceHandler.java

```diff
@@ -15,31 +15,41 @@
  */
 package ro.pippo.core.route;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 /**
  * Serves classpath resources.
  *
  * @author James Moger
  */
 public class ClasspathResourceHandler extends UrlResourceHandler {
+    private static final Logger log = LoggerFactory.getLogger(ClasspathResourceHandler.class);
 
-    private final String resourceBasePath;
+    private final Path resourceBasePath;
 
     public ClasspathResourceHandler(String urlPath, String resourceBasePath) {
         super(urlPath);
 
-        this.resourceBasePath = getNormalizedPath(resourceBasePath);
+        this.resourceBasePath = Paths.get(getNormalizedPath(resourceBasePath));
     }
 
     @Override
     public URL getResourceUrl(String resourcePath) {
-        String resourceName = getResourceBasePath() + "/" + resourcePath;
-        return this.getClass().getClassLoader().getResource(resourceName);
+        Path resolved = resourceBasePath.resolve(resourcePath).normalize();
```

Comment:
I prefer to keep `resourceBasePath` as String and to modify only `getResourceUrl`.
The modification create a `Path requestedPath` variable and check to see if this path startWith resourceBasePath (similar with https://github.com/decebals/pippo/blob/master/pippo-core/src/main/java/ro/pippo/core/route/DirectoryHandler.java#L138)